tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/RepositoryController.java

```diff
@@ -167,8 +167,15 @@ protected ModelAndView handleRequestInternal(HttpServletRequest request, HttpSer
 								.orElseThrow(() -> new HTTPException(HttpStatus.SC_BAD_REQUEST,
 										"unrecognized content type " + request.getContentType())));
 				RepositoryConfig config = RepositoryConfigUtil.getRepositoryConfig(model, repId);
```

Comment:
Kinda wish that this method couldn't return null. Throwing an exception would be best, RepositoryConfigNotFound, or at the least use optional. 

I presume it's harder to change that in a bug fix though.

Reply:
Yeah, I'd rather leave it as-is for now. Perhaps worth opening a more general refactoring issue on for 4.0 though, going through the public API and replacing some "may return null" signatures with an optional return type?